test/test_filesystem_helper.cpp

```diff
@@ -103,3 +103,53 @@ TEST(TestFilesystemHelper, is_absolute)
     }
   }
 }
+
+TEST(TestFilesystemHelper, correct_extension)
```

Comment:
it would be great to minimize the number of `is_win32` if-statements. This could be done as follow:

```c++
std::string build_example_path() {
  return is_win32 ? r"(C:\foo\bar\baz.yml)" : "/bar/foo/baz.yml";
}
```

...and then calling it in the tests directely.

Reply:
I added some helper functions, not sure the placement is the best though, also the naming could perhaps be better. Let me know what you think of those and the placement etc.